src/ynEIGEN/TokenStakingNodesManager.sol

```diff
@@ -241,6 +257,42 @@ contract TokenStakingNodesManager is AccessControlUpgradeable, ITokenStakingNode
         emit MaxNodeCountUpdated(_maxNodeCount);
     }
 
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  WITHDRAWALS  -------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    function processPrincipalWithdrawals(
```

Comment:
What if we don't do this at all in ynEigen?

The reason this was a thing in ynETH is because

1. we had to take fees on withdrawals - therefore we had to insert fee collection logic on the withdrawal path + the trusted input on how much was rewards
2  there was accounting to change for each StakingNode (allocated staked eth needed to change)
3. The principal withdrawals came out in 32 ETH chunks (no further granularity) - therefore we had to have the "withdraw" and "reinvest" breakdown

As far as i can tell 1,2,3 don't apply here, but there is some accounting inside each of the TokenStakingNode instances.

What if simply have a role there that is able to do this and is constrained to send things to the vault, and leave this contract fully unchanged?

it will act on each TokenStakingNode as it does to trigger withdrawals and queuedWithdrawals

Reply:
we may need to store just a reference to the redemptionAssetVault since the staking nodes query the Manager for data